server/src/main/java/org/opensearch/repositories/Repository.java

```diff
@@ -270,6 +309,20 @@ void restoreShard(
         ActionListener<Void> listener
     );
 
+    /**
+     * Returns Snapshot Shard Metadata for remote store interop enabled snapshot.
+     * <p>
+     * The index can be renamed on restore, hence different {@code shardId} and {@code snapshotShardId} are supplied.
+     * @param snapshotId      snapshot id
+     * @param indexId         id of the index in the repository from which the restore is occurring
+     * @param snapshotShardId shard id (in the snapshot)
+     */
+    public BlobStoreRemStoreBasedIndexShardSnapshot getRemoteStoreInteropShardMetadata(
```

Comment:
Let's not use `interop` in names. rename to something like `getRemoteStoreBackedIndexShardMetadata()`